src/Polysemy/Dispatch.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE NumDecimals     #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies    #-}
+
+module Polysemy.Dispatch where
+
+import qualified Control.Concurrent.Async as A
+import           Control.Concurrent.Chan.Unagi
+import           Control.Concurrent.MVar
+import           Control.Concurrent (threadDelay)
+import           Control.Monad
+import           Data.Maybe
+import           Polysemy
+import           Polysemy.Internal
+import           Polysemy.Internal.Union
+import           Polysemy.State
+
+
+data Request r end = forall a. Request
+  { responseMVar :: MVar (Sem '[end] a)
+  , request      :: Union r (Sem r) a
+  }
+
+dispatchEverything
+    :: ( LastMembers end r
+       , Member (Lift IO) '[end]
```

Comment:
Curiousity:

Is this equivalent to
`Lift IO ~ end`
?
Or is `Member` required for some custom type error or `Find` and `IndexOf` machinery?

Reply:
`Member` is the thing that actually lets `send` work. It's yucky, but I don't think it's a game stopper.

Reply:
When I first wrote this `LastMembers` would match any suffix of `r`, thus the name and weird style around it. It ended up not working for reasons I didn't want to debug last night, but I'm not convinced it's a fundamental limitation.